### DISCUSSION TITLE
`isConsoleRequest()` may return erroneously when testing the Web application in Codeception

### DISCUSSION THREAD

**Discussion by michaelrog** [craftcms/cms#10267] [Q&A] [answered]
In the course of plugin support, I encountered a weird issue where [Craft was unable to resolve its current route while running via Codeception, causing tests to fail](https://github.com/TopShelfCraft/New-Relic/issues/7).

The relevant logic is in a branch where `isConsoleRequest()` returns `true`, so the plugin is expecting to be dealing with `craft\console\Request` in that branch... but the app is bootstrapped as a _Web_ application, and therefore uses a `craft\web\Request`... leading to an Exception.

The confounding factor seems to be that, by default, `isConsoleRequest()` only uses `PHP_SAPI` to test its situation — which of course will reflect that we're running in the console, i.e. Codeception.

**Is there a way Craft could avoid tripping over this? — For example, maybe `craft\web\Request` could explicitly set `setIsConsoleRequest(false)` at init?**

**Comment by brandonkelly** [comment, marked as answer]
`getIsConsoleRequest()` is defined by `yii\base\Request`, so this is more of a Yii thing. Not really interested in overriding it or customizing the behavior, when the bug (if there is one here) really should be fixed in Yii.

**Reply by michaelrog** [reply]
Good point. I'll open up a thread there. (I imagine _somebody_ has had to contend with this before; maybe there's a way you're Supposed To deal with this...)

Thanks!
